### PR TITLE
Added debug log for missing post id in embedded permalink

### PR DESCRIPTION
**Pull request by harshilsharma63**
#### Summary
Adds stricter checks before accessing post permalink embed data post id, basically following whats being done in other location. I couldn't reproduce the bug llocally amd am still thinking of scenarios where this could occur, but meanwhile adding stricter checks before accessing values.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-53138

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Comment by hmhealey** [comment]
There's something really bizarre going on in this code. This part that you're changing was introduced in https://github.com/mattermost/mattermost/pull/23226 to remove embeds that contain a deleted post in `embed.data.post`, but based on the `removeUnneededMetadata` function in that file ([here](https://github.com/mattermost/mattermost/blob/master/webapp/channels/src/packages/mattermost-redux/src/reducers/entities/posts.ts#L77)), `embed.data.post` should always be deleted before a post is added to that state.

I see posts in my state on community that have that field though which is likely why the change from #23226 was done, but that seems bad since it means the post never went through `removeUnneededMetadata`. Similarly, I also see posts in my store with `post.metadata.reactions` and OpenGraph embeds with their `embed.data` set, both of which should've been removed before the posts were saved in the store.

**Comment by harshilsharma63** [comment]
@hmhealey yeah, the embed metadata should be there for everything - permalink, OG data, emojis, etc.

**Comment by harshilsharma63** [comment]
@M-ZubairAhmed I've added dbeug log. I'm not checking for dev mode as that function requires full state and its not works inside a reducer. In the interest of time, leaving it as is, and will remove before release.

**Review by M-ZubairAhmed — APPROVED** [review]
Please update title it still says  what it was doing in previously

**Comment by hmhealey** [comment]
> @hmhealey yeah, the embed metadata should be there for everything - permalink, OG data, emojis, etc.

No, we actually wanted to remove some of that data like `metadata.embed.data` for OpenGraph and  `metadata.embed.data.post` for Permalink posts since they're stored separately in different parts of the Redux store. That one line I linked in my comment seems to accidentally remove the call to `removeUnneededMetadata` for any post with embeds, so I think that's the issue there.

Regardless, we can address that later. I'll approve this so we can debug the issue on community

**Review by hmhealey — APPROVED** [review]

**Comment by hmhealey** [comment]
@harshilsharma63 I'll leave this to you to merge since I think we might want to skip cherry-picking for it

**Comment by amyblais** [comment]
> I'll leave this to you to merge since I think we might want to skip cherry-picking for it, but I'm not sure about that

We can remove the "cherry/pick approved" label if we don't want to cherry-pick this, let me know.

**Comment by harshilsharma63** [comment]
@amyblais we don't wanna cherrypick this. These are only debugging logs for now to help with finding the actual cause of the linked bug. Hopefully this will help in finding the root cause. The fix later can be cherrypicked if needed.

I've removed the cherrypick label.

**Pull request merged**
